bot/models/extensions/language/pun.py

```diff
@@ -0,0 +1,17 @@
+from sqlalchemy import Text, Column, Integer
+from sqlalchemy.orm import relationship
+from bot import app
+from db.model import Model
+
+
+class Pun(app.base, Model):
+    __tablename__ = 'puns'
+
+    id = Column(Integer, primary_key=True)
+    text = Column(Text(), unique=True)
+    pun_words = relationship("PunWord")
+
+    @property
+    def words(self):
```

Comment:
Nice use of a generator, weirdly the first "advance" thing I've seen someone use for this project XD. Well thought.